File: Beginner/isPalindrome.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False

        div = 1
        while x >= 10 * div:
            div *= 10
        
        while x:
            right = x % 10
            left = x // div

            if left != right: return False

            x = (x % div) // 10
            div = div / 100
        return True
    
num = int(input("Please enter the number: "))
s = Solution()

if s.isPalindrome(num): 
    print("The number is a palindrome")
else:
    print("The number is not a palindrome")

File: Beginner/isPalindrome.cpp
// solving beginner level palindrom leetcode problem
#include <iostream>
using namespace std;


bool isPalindrome(int x) {
    if (x < 0) {
        return false;
    }

    long reversed = 0;
    int y = x; 

    while (y > 0) {
        reversed = reversed * 10 + y % 10;
        y /= 10;
    }

    return reversed == x;
}

int main()
{    
    int n;
    cout << "Enter a positive number: ";
    cin >> n;
    
    if (isPalindrome(n)) {
        cout << "The number is a palindrome." << endl;
    } else {
        cout << "The number is not a palindrome." << endl;
    }
}